qemu/tests/rh_qemu_iotests.py

```diff
@@ -0,0 +1,155 @@
+    qemu binaries.
+
+    1) Get url of src rpm from brew
+    2) Unpack src code and apply patches
+    3) Run test for the file format detected
+    4) Check result
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def retry_command(cmd):
+        """
+        Retry the command when it fails, and raise the error once
+        retry has exceeded the maximum number
+
+        :param cmd: command expect to be executed
+        :return: output of the command
+        """
+        max_retry = int(params.get("max_retry", 3))
+        retry = max_retry
+        while retry:
+            retry -= 1
+            try:
+                return process.system_output(cmd, shell=True)
+            except process.CmdError, detail:
+                msg = "Fail to execute command"
+                logging.error("%s: %s." % (msg, detail))
+        raise exceptions.TestError("%s after %s times retry: %s" %
+                                   (msg, max_retry, detail))
+
+    def get_qemu_src_rpm_url():
+        """
+        Check the version of qemu-kvm on host, then get url
+        of qemu Source RPM.
+
+        :return: url of qemu source rpm
+        """
+        version_pattern = params.get("qemu_version_pattern")
+        src_rpm_pattern = params.get("src_rpm_pattern", ".*src.rpm")
+        download_root = params.get("download_root_url")
+        get_qemu_version_cmd = ("%s -version" %
+                                utils_misc.get_qemu_binary(params))
+        output = process.system_output(get_qemu_version_cmd, shell=True)
+        qemu_version = re.search(version_pattern, output, re.M).group(0)
+        cmd = "brew buildinfo %s" % qemu_version
+        buildinfo = retry_command(cmd)
+        try:
+            src_brew_link = re.findall(src_rpm_pattern, buildinfo, re.I)[0]
+        except IndexError:
+            raise exceptions.TestError("Could not get qemu src package "
+                                       "brew link matching pattern '%s'" %
+                                       src_rpm_pattern)
+        url = download_root + src_brew_link[11:]
+        logging.info("Brew URL is %s" % url)
+        return url
+
+    def get_qemu_src_from_rpm(rpmbuild_dir, url):
+        """
+        Unpacking src code and applying any patches
+
+        :param rpmbuild_dir: build path of rpmbuild
+        :param url: url of qemu source rpm
+        :return: directory of qemu source code
+        """
+        cmd = "rpm -ivhf %s" % url
+        retry_command(cmd)
+        spec = os.path.join(rpmbuild_dir, 'SPECS/qemu-kvm.spec')
+        BUILD_dir = os.path.join(rpmbuild_dir, 'BUILD')
+        cmd = "rpmbuild -bp %s --nodeps" % spec
+        process.system_output(cmd, shell=True)
+        src_dir = os.listdir(BUILD_dir)[0]
+        return os.path.join(BUILD_dir, src_dir)
+
+    def run_configure_script(qemu_src_dir):
```

Comment:
I guest it only need to do after a version, better to use package version as check condition